pgmpy/readwrite/UAI.py

```diff
@@ -0,0 +1,256 @@
+        elif string:
+            self.network = string
+        else:
+            raise ValueError("Must specify either path or string.")
+        self.grammar = self.get_grammar()
+        self.network_type = self.get_network_type()
+        self.variables = self.get_variables()
+        self.domain = self.get_domain()
+        self.edges = self.get_edges()
+        self.tables = self.get_tables()
+
+    def get_grammar(self):
+        """
+        Returns the grammar of the UAI file.
+        """
+        network_name = Word(alphas).setResultsName('network_name')
+        no_variables = Word(nums).setResultsName('no_variables')
+        grammar = network_name+no_variables
+        self.no_variables = int(grammar.parseString(self.network)['no_variables'])
+        domain_variables = (Word(nums)*self.no_variables).setResultsName('domain_variables')
+        grammar += domain_variables
+        no_functions = Word(nums).setResultsName('no_functions')
+        grammar += no_functions
+        self.no_functions = int(grammar.parseString(self.network)['no_functions'])
+        for function in range(0, self.no_functions):
+            scope_grammar = Word(nums).setResultsName('fun_scope_'+str(function))
+            grammar += scope_grammar
+            function_scope = grammar.parseString(self.network)['fun_scope_'+str(function)]
+            function_grammar = ((Word(nums))*int(function_scope)).setResultsName('fun_'+str(function))
+            grammar += function_grammar
+
+        floatnumber = Combine(Word(nums) + Optional(Literal(".") + Optional(Word(nums))))
+        for function in range(0, self.no_functions):
+            no_values_grammar = Word(nums).setResultsName('fun_no_values_' + str(function))
+            grammar += no_values_grammar
+            no_values = grammar.parseString(self.network)['fun_no_values_' + str(function)]
+            values_grammar = ((floatnumber)*int(no_values)).setResultsName('fun_values_'+str(function))
+            grammar += values_grammar
+        return grammar
+
+    def get_network_type(self):
+        """
+        Returns the type of network defined by the file.
+
+        Returns
+        -------
+        string : str
+            String containing network type.
+
+        Example
+        -------
+        >>> reader = UAIReader('TestUAI.uai')
+        >>> reader.get_network_type()
+        'MARKOV'
+        """
+        network_type = self.grammar.parseString(self.network)
+        return network_type['network_name']
+
+    def get_variables(self):
+        """
+        Returns a list of variables.
+        Each variable is represented by an index of list.
+        For example if the no of variables are 4 then the list will be
+        [var_0, var_1, var_2, var_3]
+
+        Returns
+        -------
+        list: list of variables
+
+        Example
+        -------
+        >>> reader = UAIReader('TestUAI.uai')
+        >>> reader.get_variables()
+        ['var_0', 'var_1', 'var_2']
```

Comment:
@vivek425ster You should be mentioning this somewhere in the `get_model` method that the variable names would be in this pattern.